packages/qulacs/quri_parts/qulacs/simulator.py

```diff
@@ -44,11 +46,13 @@ def run_circuit(
         raise ValueError("Inconsistent qubit length between circuit and state")
 
     qulacs_state = ql.QuantumState(circuit.qubit_count)
-    qulacs_state.load(init_state)
+    qulacs_state.load(cast_to_list(init_state))
 
     qulacs_cicuit = convert_circuit(circuit)
     qulacs_cicuit.update_quantum_state(qulacs_state)
 
-    new_state_vector: NDArray[cfloat] = qulacs_state.get_vector()
+    # The `.get_matrix()` method returns NDArray[complex128, _Shape]
+    # (_Shape should be the first argument)
```

Comment:
I think this is clearer:
```suggestion
    # We need to disable type check due to an error in qulacs type annotation
    # https://github.com/qulacs/qulacs/issues/537
```